src/read.c

```diff
@@ -799,6 +799,15 @@ static avifResult avifMetaFindOrCreateItem(avifMeta * meta, uint32_t itemID, avi
         }
     }
 
+    if (meta->items.count != 0) {
+        // ISO/IEC 23008-12, First edition, 2017-12, Section 9.3.1:
+        //   Each ItemPropertyAssociation box shall be ordered by increasing item_ID, and there shall
+        //   be at most one association box for each item_ID, in any ItemPropertyAssociation box.
+        const uint32_t lastID = meta->items.item[meta->items.count - 1]->id;
+        avifDiagnosticsPrintf(diag, "The added itemID [%u] does not preserve the itemID order", itemID);
+        AVIF_CHECKERR(itemID > lastID, AVIF_RESULT_BMFF_PARSE_FAILED);
```

Comment:
IMPORTANT: Strictly speaking, what you do is correct. Like `errno`, the diagnostic message should only be used when a function fails. So it is fine to log a diagnostic message before the check for error. But ideally we should log a diagnostic message only when a function fails.

To do this, we cannot use the `AVIF_CHECKERR` macro. We need to use an if statement:

```
        if (itemID <= lastID) {
            avifBreakOnError();
            avifDiagnosticsPrintf(diag, "The added itemID [%u] does not preserve the itemID order", itemID);
            return AVIF_RESULT_BMFF_PARSE_FAILED;
        }
```

Reply:
Cleaner indeed, done